src/gluonts/time_feature/holiday.py

```diff
@@ -78,44 +78,25 @@ def distance_to_day(index):
     offset=[pd.DateOffset(weekday=TH(4)), Day(4)],
 )
 
-NEW_YEARS_DAY = "new_years_day"
-MARTIN_LUTHER_KING_DAY = "martin_luther_king_day"
-SUPERBOWL = "superbowl"
-PRESIDENTS_DAY = "presidents_day"
-GOOD_FRIDAY = "good_friday"
-EASTER_SUNDAY = "easter_sunday"
-EASTER_MONDAY = "easter_monday"
-MOTHERS_DAY = "mothers_day"
-INDEPENDENCE_DAY = "independence_day"
-LABOR_DAY = "labor_day"
-MEMORIAL_DAY = "memorial_day"
-COLUMBUS_DAY = "columbus_day"
-THANKSGIVING = "thanksgiving"
-CHRISTMAS_EVE = "christmas_eve"
-CHRISTMAS_DAY = "christmas_day"
-NEW_YEARS_EVE = "new_years_eve"
-BLACK_FRIDAY = "black_friday"
-CYBER_MONDAY = "cyber_monday"
-
 SPECIAL_DATE_FEATURES = {
-    NEW_YEARS_DAY: distance_to_holiday(NewYearsDay),
-    MARTIN_LUTHER_KING_DAY: distance_to_holiday(USMartinLutherKingJr),
-    SUPERBOWL: distance_to_holiday(SuperBowl),
-    PRESIDENTS_DAY: distance_to_holiday(USPresidentsDay),
-    GOOD_FRIDAY: distance_to_holiday(GoodFriday),
-    EASTER_SUNDAY: distance_to_holiday(EasterSunday),
-    EASTER_MONDAY: distance_to_holiday(EasterMonday),
-    MOTHERS_DAY: distance_to_holiday(MothersDay),
-    INDEPENDENCE_DAY: distance_to_holiday(IndependenceDay),
-    LABOR_DAY: distance_to_holiday(USLaborDay),
-    MEMORIAL_DAY: distance_to_holiday(USMemorialDay),
-    COLUMBUS_DAY: distance_to_holiday(USColumbusDay),
-    THANKSGIVING: distance_to_holiday(USThanksgivingDay),
-    CHRISTMAS_EVE: distance_to_holiday(ChristmasEve),
-    CHRISTMAS_DAY: distance_to_holiday(ChristmasDay),
-    NEW_YEARS_EVE: distance_to_holiday(NewYearsEve),
-    BLACK_FRIDAY: distance_to_holiday(BlackFriday),
-    CYBER_MONDAY: distance_to_holiday(CyberMonday),
+    "new_years_day": distance_to_holiday(NewYearsDay),
```

Comment:
Some of them are referenced in the docstings below.

Reply:
I think we can remove that class entirely, it's only used in tests and I don't see any way the user can use it with estimators.

Instead, holidays should probably work just like any other `TimeFeature`?

Reply:
so if they are time features we just need to make sure they are not called on the fly as these operations are slow and in theory are just need once and then every time series in a dataset can use them...

Reply:
Good point @kashif, some caching would help there I guess. In general I believe we should re-think how these features are specified to estimators, since now there seems to be too much going on.